custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,421 @@
+                if entry.get("flags").get("unverified"):
+                    logger.warning(f"This entry '{entry_id}' is unverified, subtitles might be incomplete or have quality issues!")
+                
+                break
+            
+        if entry_property_value is None:
+            logger.error('Matched no entries.')
+            return None
+        
+        # Get a list of subtitles for entry
+        retry_count = 0
+        while True:
+            episode_number = video.episode
+            url = f"entries/{entry_id}/files?episode={episode_number}"
+            data = self._get_jimaku_response(url)
+            
+            # Edge case: When dealing with a cour, episodes could be uploaded with their episode numbers having an offset applied
+            if not data and self.video_episode_number_override and retry_count < 1:
+                episode_number = self.video_episode_number_override
+                logger.warning(f"Found no subtitles for {episode_number}, but will retry with offset-adjusted episode number {self.video_episode_number_override}.")
+                retry_count += 1
+            elif not data:
+                return None
+            else:
+                break
+        
+        # Filter subtitles
+        list_of_subtitles = []
+        
+        archive_formats_blacklist = (".7z",) # Unhandled format
+        if not self.enable_archives:
+            disabled_archives = (".zip", ".rar")
+            
+            # Handle shows that only have archives uploaded
+            filter = [item for item in data if not item['name'].endswith(disabled_archives)]
+            if len(filter) == 0:
+                logger.warning("Archives are disabled, but only archived subtitles have been returned. Will therefore download anyway.")
+            else:
+                archive_formats_blacklist += disabled_archives
+
+        # Order subtitles to maximize quality
+        # Certain subtitle sources, such as Netflix/WebRip usually yield superior quality over others
+        sorted_data = sorted(data, key=self._subtitle_sorting_key)
+
+        for item in sorted_data:
+            subtitle_filename = item.get('name')
+            subtitle_filesize = item.get('size')
+            subtitle_url = item.get('url')
+            
+            if not self.enable_ai_subs:
+                #ToDo: Implement
+                # Observed tags: {ai whisper generated} [Generated by whisper]
+                for ai_keyword in ["generated", "whisper"]:
+                    if re.search(r'\b' + re.escape(ai_keyword) + r'\b', subtitle_filename.lower()):
+                        logger.warning(f"Skipping AI generated subtitle '{subtitle_filename}'")
+                        continue
+            
+            # Check if file is corrupt; The average file stands at ~20kB, we'll set the threshold at 5kB though 
+            if subtitle_filesize < 5000:
```

Comment:
Is this will be the case for FORCED subtitles too? 🤔

Reply:
I've not yet seen isolated forced subs on Jimaku so I don't have a size reference.  
I take you've seen such subs smaller than 5k then? (Not necessarily only on Jimaku)

Perhaps we could set `subtitle_filesize` even lower as to only catch obviously corrupted files.

Reply:
I need to dig Jimaku, it might be worthy to ask on their Discord but this is an example

https://animetosho.org/view/zerobuild-frieren-beyond-journeys-end-s01e25-web-1080p.n1790008

Check the `[English Forced [CR] [eng, SRT]]`

Reply:
Its a totally valid subtitle, 1KB zipped and 4KB unzipped.

Reply:
I lowered the threshold to just 500 bytes, this will only catch (very) likely corrupted files but should still allow forced subs.